src/base/action.lua

```diff
@@ -257,6 +257,47 @@
 		return false
 	end
 
+---
+-- Determines if an action supports a particular toolset.
+--
+-- @param language
+--    The language that toolset belongs.
+-- @param toolset
+--    The toolset to check.
+-- @returns
+--    True if the toolset is supported, false otherwise.
+---
+
+	function action.supportsToolset(language, toolset)
+		local function buildLanguageKeysMap()
```

Comment:
Not sure how it should be done with extern registered languages. Apply to `valid_tools` entry or registered action anyway...